alameda.js

```diff
@@ -111,10 +111,10 @@ var requirejs, require, define;
         shim: {},
         config: {}
       },
-      mapCache = {},
+      mapCache = Object.create(null),
       requireDeferreds = [],
-      deferreds = {},
-      calledDefine = {},
+      deferreds = Object.create(null),
+      calledDefine = Object.create(null),
       calledPlugin = {},
```

Comment:
Replacing hasProp with `prop in obj` made the timeline view / profiler dramatically easier to read, billing time to the right higher-level functions.

Whether it affected load times or not is harder to say - I don't have the right benchmarks in place to measure small changes.

If you think a smaller utility function to wrap Object.create(null) is justified, that's fine with me.